gateway/cert.go

```diff
@@ -328,7 +352,16 @@ func (gw *Gateway) getTLSConfigForClient(baseConfig *tls.Config, listenPort int)
 
 	return func(hello *tls.ClientHelloInfo) (*tls.Config, error) {
 		if config, found := tlsConfigCache.Get(hello.ServerName + listenPortStr); found {
-			return config.(*tls.Config).Clone(), nil
+			newConfig := config.(*tls.Config).Clone()
+
+			if gwConfig.HttpServerOptions.SkipClientCAAnnouncement {
```

Comment:
@buger, why is this modification of the cached TLS config is necessary?

Reply:
You need clone if for each client, otherwise you will get some race issues.

Reply:
@buger agreed with the `clone` part but my point is that if we save the `tlsConfig` with the `VerifyPeerCertificate`, `ClientCAs` and `ClientAuth` already set - we won't need to set this twice